In [1]:
import sys
sys.path.append('../')

import Dataset_load
from gammapy.modeling.models import Models

In [2]:
c = Dataset_load.load_config()
awo, aw, ewo, ew = c['_colors'] 
folder = c['folder']
livetime  = c['livetime']

In [3]:
path = f'../Combined100_PKSflare_crab_cutoff/data/0_model_nui_livetime_{livetime}_np.yml'
model_combined = Models.read(path)

path = f'../Eff_area100_PKSflare_crab_cutoff/data/0_model_nui_livetime_{livetime}_np.yml'
model_eff = Models.read(path)

path = f'../E_reco_PKSflare_crab_cutoff/data/0_model_nui_livetime_{livetime}_np.yml'
model_ereco = Models.read(path)

models = [model_eff, model_ereco, model_combined]

In [4]:
path = f'../Combined100_PKSflare_crab_cutoff/data/0_model_livetime_{livetime}_np.yml'
model_combined_no = Models.read(path)

path = f'../Eff_area100_PKSflare_crab_cutoff/data/0_model_livetime_{livetime}_np.yml'
model_eff_no = Models.read(path)

path = f'../E_reco_PKSflare_crab_cutoff/data/0_model_livetime_{livetime}_np.yml'
model_ereco_no = Models.read(path)

models_no = [model_eff_no, model_ereco_no, model_combined_no]

In [5]:
print(model_combined[0])
print(model_eff[0])
print(model_ereco[0])

SkyModel

  Name                      : U64xP7x-
  Datasets names            : None
  Spectral model type       : ExpCutoffPowerLawSpectralModel
  Spatial  model type       : PointSpatialModel
  Temporal model type       : 
  Parameters:
    index                      :      2.299  +/-    0.13             
    amplitude                  :   3.85e-11  +/- 6.0e-12 1 / (cm2 s TeV)
    reference  (frozen)        :      1.000      TeV         
    lambda_                    :      0.100  +/-    0.04 1 / TeV     
    alpha      (frozen)        :      1.000                  
    lon_0                      :    329.680  +/-    0.00 deg         
    lat_0                      :    -30.222  +/-    0.00 deg         


SkyModel

  Name                      : 92gHQ4qx
  Datasets names            : None
  Spectral model type       : ExpCutoffPowerLawSpectralModel
  Spatial  model type       : PointSpatialModel
  Temporal model type       : 
  Parameters:
    index                      :      2.302  

## Index

In [6]:
for m in models_no:
    par = m.parameters['index']
    print(par.name)
    try:
        print(f'{par.value:.4}  \pm  {par.error:.4} -{par.error_n:.4}  +{par.error_p:.4}')
    except:
        print(f'{par.value:.4}  \pm  {par.error:.4} ')
        

index
2.3  \pm  0.09232 -0.09572  +0.09008
index
2.227  \pm  0.08819 -0.09584  +0.0902
index
2.3  \pm  0.09232 -0.09584  +0.0902


In [7]:
for m in models:
    par = m.parameters['index']
    print(par.name)
    try:
        print(f'{par.value:.4}  \pm  {par.error:.4} -{par.error_n:.4}  +{par.error_p:.4}')
    except:
        print(f'{par.value:.4}  \pm  {par.error:.4} ')
        

index
2.302  \pm  0.09224 -0.09838  +0.09174
index
2.299  \pm  0.1291 -0.1282  +0.1503
index
2.299  \pm  0.1306 -0.13  +0.1512


## amplitude

In [8]:
for m in models_no:
    par = m.parameters['amplitude']
    print(par.name)
    try:
        print(f'{par.value:.4}  \pm  {par.error:.4} -{par.error_n:.4}  +{par.error_p:.4}')
    except:
        print(f'{par.value:.4}  \pm  {par.error:.4} ')
        

amplitude
3.85e-11  \pm  2.59e-12 -2.431e-12  +2.808e-12
amplitude
4.057e-11  \pm  2.502e-12 -2.431e-12  +2.808e-12
amplitude
3.85e-11  \pm  2.59e-12 -2.431e-12  +2.809e-12


In [9]:
for m in models:
    par = m.parameters['amplitude']
    print(par.name)
    try:
        print(f'{par.value:.4}  \pm  {par.error:.4} -{par.error_n:.4}  +{par.error_p:.4}')
    except:
        print(f'{par.value:.4}  \pm  {par.error:.4} ')
        

amplitude
3.849e-11  \pm  5.934e-12 -5.334e-12  +6.439e-12
amplitude
3.852e-11  \pm  2.607e-12 -2.578e-12  +3.051e-12
amplitude
3.853e-11  \pm  5.967e-12 -5.407e-12  +6.566e-12


## lambda_

In [10]:
for m in models_no:
    par = m.parameters['lambda_']
    print(par.name)
    try:
        print(f'{par.value:.4}  \pm  {par.error:.4} -{par.error_n:.4}  +{par.error_p:.4}')
    except:
        print(f'{par.value:.4}  \pm  {par.error:.4} ')
        

lambda_
0.1  \pm  0.03995 -0.03707  +0.04356
lambda_
0.138  \pm  0.03805 -0.03707  +0.04356
lambda_
0.1  \pm  0.03995 -0.03706  +0.0436


In [11]:
for m in models:
    par = m.parameters['lambda_']
    print(par.name)
    try:
        print(f'{par.value:.4}  \pm  {par.error:.4} -{par.error_n:.4}  +{par.error_p:.4}')
    except:
        print(f'{par.value:.4}  \pm  {par.error:.4} ')
        

lambda_
0.09952  \pm  0.03891 -0.03695  +0.04345
lambda_
0.09988  \pm  0.03984 -0.04084  +0.04653
lambda_
0.09988  \pm  0.03984 -0.04095  +0.04644


In [18]:
headline = "\\textbf{Point Source} & &  $\Phi_0\ [10^{-11}\, \\text{cm}^{-2}\, \\text{s}^{-1}\,\\text{TeV}^{-1}]$  & $\Gamma$ & $\lambda = 1/E_c\ [\\text{TeV}^{-1}]$ \\\ \hline "
input_ = ' & Simulation input & $3.85 $  & $2.30 $  & $0.10 $  \\\  \hline'

par = models_no[0].parameters['amplitude']
without = rf' &Without systematic & \error {{{par.value*1e11:.3}}} {{{par.error*1e11:.3}}}  {{{par.error_p*1e11:.3}}}  {{{par.error_n*1e11:.3}}} '
par = models_no[0].parameters['index']
without += rf' & \error {{{par.value:.3}}} {{{par.error:.3}}}  {{{par.error_p:.3}}}  {{{par.error_n:.3}}} '
par = models_no[0].parameters['lambda_']
without += rf' & \error  {{{par.value:.3}}} {{{par.error:.3}}}  {{{par.error_p:.3}}}  {{{par.error_n:.3}}}  \\  \hline'


par = models[0].parameters['amplitude']
eff = rf' Effective Area & With fitting  & \error {{{par.value*1e11:.3}}} {{{par.error*1e11:.3}}}  {{{par.error_p*1e11:.3}}}  {{{par.error_n*1e11:.3}}} '
par = models[0].parameters['index']
eff += rf' & \error  {{{par.value:.3}}} {{{par.error:.3}}}  {{{par.error_p:.3}}}  {{{par.error_n:.3}}} '
par = models[0].parameters['lambda_']
eff += rf' & \error  {{{par.value:.3}}} {{{par.error:.3}}}  {{{par.error_p:.3}}}  {{{par.error_n:.3}}}  \\  \hline'

par = models[1].parameters['amplitude']
bias = rf' Bias & With fitting  & \error {{{par.value*1e11:.3}}} {{{par.error*1e11:.3}}}  {{{par.error_p*1e11:.3}}}  {{{par.error_n*1e11:.3}}} '
par = models[1].parameters['index']
bias += rf' & \error  {{{par.value:.3}}} {{{par.error:.3}}}  {{{par.error_p:.3}}}  {{{par.error_n:.3}}} '
par = models[1].parameters['lambda_']
bias += rf' & \error  {{{par.value:.3}}} {{{par.error:.3}}}  {{{par.error_p:.3}}}  {{{par.error_n:.3}}}  \\  \hline'

par = models[2].parameters['amplitude']
bias_eff = rf' Effectiva Area \& Bias & With fitting  & \error {{{par.value*1e11:.3}}} {{{par.error*1e11:.3}}}  {{{par.error_p*1e11:.3}}}  {{{par.error_n*1e11:.3}}} '
par = models[2].parameters['index']
bias_eff += rf' & \error  {{{par.value:.3}}} {{{par.error:.3}}}  {{{par.error_p:.3}}}  {{{par.error_n:.3}}} '
par = models[2].parameters['lambda_']
bias_eff += rf' & \error {{{par.value:.3}}} {{{par.error:.3}}}  {{{par.error_p:.3}}}  {{{par.error_n:.3}}}  \\  \hline'


print(headline)
print(input_)
print()
print(without)
print()
print(eff)
print()
print(bias)
print()
print(bias_eff)



\textbf{Point Source} & &  $\Phi_0\ [10^{-11}\, \text{cm}^{-2}\, \text{s}^{-1}\,\text{TeV}^{-1}]$  & $\Gamma$ & $\lambda = 1/E_c\ [\text{TeV}^{-1}]$ \\ \hline 
 & Simulation input & $3.85 $  & $2.30 $  & $0.10 $  \\  \hline

 &Without systematic & \error {3.85} {0.259}  {0.281}  {0.243}  & \error {2.3} {0.0923}  {0.0901}  {0.0957}  & \error  {0.1} {0.0399}  {0.0436}  {0.0371}  \\  \hline

 Effective Area & With fitting  & \error {3.85} {0.593}  {0.644}  {0.533}  & \error  {2.3} {0.0922}  {0.0917}  {0.0984}  & \error  {0.0995} {0.0389}  {0.0435}  {0.0369}  \\  \hline

 Bias & With fitting  & \error {3.85} {0.261}  {0.305}  {0.258}  & \error  {2.3} {0.129}  {0.15}  {0.128}  & \error  {0.0999} {0.0398}  {0.0465}  {0.0408}  \\  \hline

 Effectiva Area \& Bias & With fitting  & \error {3.85} {0.597}  {0.657}  {0.541}  & \error  {2.3} {0.131}  {0.151}  {0.13}  & \error {0.0999} {0.0398}  {0.0464}  {0.041}  \\  \hline


     \textbf{Point Source} & &  $\Phi_0\ [10^{-11}\, \text{cm}^{-2}\, \text{s}^{-1}\,\text{TeV}^{-1}]$  & $\Gamma$ & $\lambda = 1/E_c\ [\text{TeV}^{-1}]$ \\ \hline 

     & Simulation input & $3.85 $  & $2.30 $  & $0.10 $  \\  \hline
       &Without systematic & \error {3.85} { 0.25895}  {0.281}  {0.243} & \error {2.30} {0.092}  {0.09007}  {0.0957} & \error {0.10} { 0.04}  {0.0436}  {0.0371} \\  \hline

     Effective Area & With fitting  & \error {3.85} { 0.593}  {0.643}  {0.533} & \error {2.30} { 0.092}  {0.0917}  {0.0983} & \error {0.10} {0.0389}  {0.043}  {0.0369} \\  \hline

       Bias & With fitting  & \error {3.85} { 0.262}  {0.305}  {0.258} & \error {2.30} { 0.135}  {0.15}  {0.128} & \error {0.10} { 0.0401}  {0.0438}  {0.0371} \\  \hline 

    Effectiva Area \& Bias &With fitting & \error {3.85} { 0.593}  {0.517}  {0.445} & \error {2.30} { 0.116}  {0.15}  {0.128} & \error {0.10} { 0.0412}  {0.046}  {0.041}\